flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -467,6 +487,128 @@ public void registerAtResourceManager(final String address) {
 		//TODO:: register at the RM
 	}
 
+	@RpcMethod
+	public NextInputSplit requestNextInputSplit(final JobVertexID vertexID, final ExecutionAttemptID executionAttempt) {
```

Comment:
Thanks for your comments @tillrohrmann . It all sounds reasonable to me. Limit the rpc error within one specified Exception is a good idea, thus user can easily know what went wrong. 
One minor suggestion, may be we should create a base class `RpcException`, and make something like `RpcConnectionException`, `RpcExecutionException` and `RpcTimeoutException` to inherit from that, to make a better more clear. This can be done in another jira, though.